exasol/python_extension_common/deployment/temp_schema.py

```diff
@@ -50,11 +50,11 @@ def temp_schema(conn: pyexasol.ExaConnection,
     conn                - pyexasol connection.
     schema_name_length  - Number of characters in the temporary schema name.
     """
-    current_schema = _get_schema(conn)
+    current_schema = get_schema(conn)
     schema = ''
     try:
         schema = _create_random_schema(conn, schema_name_length)
         yield schema
     finally:
-        _delete_schema(conn, schema)
-        _set_schema(conn, current_schema)
+        delete_schema(conn, schema)
+        set_schema(conn, current_schema)
```

Comment:
Here is my proposal: 

```python
@contextmanager
def temp_schema(
        conn: pyexasol.ExaConnection,
        schema_name_length: int = 20,
        reuse: bool = False,
) -> Generator[str, None, None]:
    """
    A context manager for running an operation in a newly created
    temporary schema or the currently selected one, if it's OK, to reuse that.
    If a temporary schema is created, then it will be deleted after the
    operation is competed.
    Note, that all objects created in this schema will be deleted with it.
    The context returns the name of the created or reused schema.

    conn                - pyexasol connection.
    schema_name_length  - Number of characters in the temporary schema name.
    reuse               - If a schema is currently selected, and reuse is True,
                          then do not create a temporary schema.
    """
    current_schema = get_schema(conn)
    if current_schema and reuse:
        yield current_schema
        return

    schema = ''
    try:
        schema = _create_random_schema(conn, schema_name_length)
        yield schema
    finally:
        delete_schema(conn, schema)
        set_schema(conn, current_schema)
```

Reply:
I guess in this particular scenario it will save a little bit of code, but it would make the logic of the `temp_schema` confusing. This function function is used elsewhere. Its purpose is to allow the creation of temporary objects in a temporary schema that will subsequently be deleted with the schema. I would rather leave it as it is to keep this purpose clear.